### PR TITLE
Fix the bug where the namespace cannot be deleted

### PR DESCRIPTION
**Pull request by zhengxiexie**
Fix the bug where the namespace cannot be deleted due to the username of the namespace deletion.
Use global log. 

Test Done:
On VC UI, namespace can be created and deleted succesfully.

### CODE REVIEW

File: pkg/controllers/namespace/namespace_controller.go [hunk 1]
```diff
@@ -132,6 +132,33 @@ func (r *NamespaceReconciler) createDefaultSubnetSet(ns string, defaultSubnetSiz
...
+			err := r.Client.Get(context.Background(), client.ObjectKey{
+				Namespace: ns,
+				Name:      name,
+			}, obj)
+			if err != nil {
+				return client.IgnoreNotFound(err)
+			}
+			log.Info("delete default SubnetSet", "Namespace", ns, "Name", name)
+			return r.Client.Delete(context.Background(), obj)
+		}); err != nil {
```
Comment (TaoZou1) [inline pkg/controllers/namespace/namespace_controller.go:154]: do you need to delete it from the store?

Comment (zhengxiexie) [reply pkg/controllers/namespace/namespace_controller.go:154]: It initiates a delete event, but the actual deletion occurs during the subnet set reconciliation process.

File: pkg/controllers/subnetset/subnetset_webhook.go [hunk 2]
```diff
@@ -65,17 +61,17 @@ func (v *SubnetSetValidator) Handle(ctx context.Context, req admission.Request)
...
 	} else {
 		err := v.decoder.Decode(req, subnetSet)
 		if err != nil {
-			subnetsetlog.Error(err, "error while decoding SubnetSet", "SubnetSet", req.Namespace+"/"+req.Name)
+			log.Error(err, "error while decoding SubnetSet", "SubnetSet", req.Namespace+"/"+req.Name)
 			return admission.Errored(http.StatusBadRequest, err)
 		}
 	}
-	subnetsetlog.Info("request user-info", "name", req.UserInfo.Username)
+	log.Info("request user-info", "name", req.UserInfo.Username)
```
Comment (lxiaopei) [inline pkg/controllers/subnetset/subnetset_webhook.go:74]: remove this debug log？

Comment (zhengxiexie) [reply pkg/controllers/subnetset/subnetset_webhook.go:74]: ok

File: pkg/controllers/namespace/namespace_controller.go [hunk 3]
```diff
@@ -132,6 +132,33 @@ func (r *NamespaceReconciler) createDefaultSubnetSet(ns string, defaultSubnetSiz
...
+		if err := retry.OnError(retry.DefaultRetry, func(err error) bool {
+			return err != nil
+		}, func() error {
+			obj := &v1alpha1.SubnetSet{}
+			err := r.Client.Get(context.Background(), client.ObjectKey{
+				Namespace: ns,
+				Name:      name,
+			}, obj)
+			if err != nil {
+				return client.IgnoreNotFound(err)
```
[the remark this answers is not in the document]
Comment (zhengxiexie) [reply pkg/controllers/namespace/namespace_controller.go:150]: if err when getting subnetset, it is not always "NotFound" error right?
yes, will retry
what happened for this IgnoreNotFound error?
if it is really NotFound, we ignore it and not delete, which is as expected

[the remark this answers is not in the document]
Comment (lxiaopei) [reply pkg/controllers/namespace/namespace_controller.go:150]: if err != nil {
				return client.IgnoreNotFound(err)

we cannot know the err is NotFound error, if failed to connect to k8s api server, for Get, it also throws error but here return "client.IgnoreNotFound(err)"

[the remark this answers is not in the document]
Comment (zhengxiexie) [reply pkg/controllers/namespace/namespace_controller.go:150]: The definition is 
```
func IgnoreNotFound(err error) error {
	if apierrors.IsNotFound(err) {
		return nil
	}
	return err
}
```

[the remark this answers is not in the document]
Comment (lxiaopei) [reply pkg/controllers/namespace/namespace_controller.go:150]: OK

File: pkg/controllers/subnetset/subnetset_webhook.go [hunk 4]
```diff
@@ -65,17 +61,17 @@ func (v *SubnetSetValidator) Handle(ctx context.Context, req admission.Request)
...
 	} else {
 		err := v.decoder.Decode(req, subnetSet)
 		if err != nil {
-			subnetsetlog.Error(err, "error while decoding SubnetSet", "SubnetSet", req.Namespace+"/"+req.Name)
+			log.Error(err, "error while decoding SubnetSet", "SubnetSet", req.Namespace+"/"+req.Name)
 			return admission.Errored(http.StatusBadRequest, err)
 		}
 	}
-	subnetsetlog.Info("request user-info", "name", req.UserInfo.Username)
+	log.V(1).Info("request user-info", "name", req.UserInfo.Username)
```
Comment (lxiaopei) [inline pkg/controllers/subnetset/subnetset_webhook.go:74]: remove？

**Review by TaoZou1 — APPROVED** [review]

**Review by lxiaopei — APPROVED** [review]

**Pull request merged**